cranelift/codegen/src/isa/aarch64/inst/imms.rs

```diff
@@ -106,6 +106,48 @@ impl SImm7Scaled {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub struct FPULeftShiftImm {
+    pub value: u8,
+    pub bits: u8,
+}
+
+impl FPULeftShiftImm {
+    pub fn maybe_from_u8(value: u8, bits: u8) -> Option<Self> {
+        debug_assert!(bits == 32 || bits == 64);
+        if value < bits {
+            Some(FPULeftShiftImm { value, bits })
+        } else {
+            None
+        }
+    }
+
+    pub fn enc(&self) -> u32 {
+        u32::from(self.value + self.bits)
```

Comment:
A comment here might be useful, because it's pretty subtle: it seems that 32-bit / 64-bit constants are indicated by bits 5 / 6, and the literal values `32` and `64` just happen to have those bits, so the add works? Also perhaps add a note that (by construction) `value` is less than `bits`, so it won't carry?